file_changed_command_runner/callback_server.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import threading
+
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+
+log = logging.getLogger(__name__)
+
+
+class CallbackHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200, "OK")
+        self.end_headers()
+        self.close_connection = True  # pylint: disable=attribute-defined-outside-init
+
+        self.server: "CallbackServer"
+        self.server.process_callback(self.path.lstrip("/"))
+
+
+class CallbackServer(HTTPServer):
+    """Listens on localhost for a request with a specific token"""
+
+    port_range = range(50000, 51000)
+
+    def __init__(self):
+        self.allow_reuse_address: bool = False
+        self._token: str = ""
+        self._event: threading.Event = threading.Event()
+
+        for port in self.port_range:
+            try:
+                super().__init__(("localhost", port), CallbackHandler)
```

Comment:
Might be slightly preferable to use 127.0.0.1 in case a user has their `/etc/hosts` set up in some messed up way (e.g. binding localhost to 0.0.0.0 instead, which could have security implications).

Reply:
good point, changed.